UniWatch/DataAccess/ClassManager.cs

```diff
@@ -0,0 +1,206 @@
+    {
+        private bool disposed = false;
+        private AppDbContext _db;
+
+        /// <summary>
+        /// Creates a default ClassManager
+        /// </summary>
+        public ClassManager()
+        {
+            _db = new AppDbContext();
+        }
+
+        /// <summary>
+        /// Creates a ClassManager with the given context
+        /// </summary>
+        /// <param name="context">The context to create the manager with</param>
+        public ClassManager(AppDbContext context)
+        {
+            _db = context;
+        }
+
+        /// <summary>
+        /// Creates a new class
+        /// </summary>
+        /// <param name="name">The name/title of the class</param>
+        /// <param name="number">The course number</param>
+        /// <param name="section">The section</param>
+        /// <param name="semester">The semester e.g. Fall 2016</param>
+        /// <param name="teacher">The teacher for this class</param>
+        /// <returns>The created class</returns>
+        /// <remarks>Throws InvalidOperationException if class already exists with the given details</remarks>
+        public Class CreateClass(string name, int number, string section, Semester semester, int year, int teacherId)
+        {
+            var existing = _db.Classes
+                .Where(c => c.Name == name &&
+                        c.Number == number &&
+                        c.Section == section &&
+                        c.Semester == semester &&
+                        c.Teacher.Id == teacherId)
+                .Count();
+
+            var teacher = _db.Teachers.Find(teacherId);
+
+            if(existing > 0 || teacher == null)
+                throw new InvalidOperationException("Error creating class");
+
+            var newClass = new Class
+            {
+                Name = name,
+                Number = number,
+                Section = section,
+                Semester = semester,
+                Teacher = teacher
+            };
+
+            var added = _db.Classes.Add(newClass);
+            _db.SaveChanges();
+            return added;
+        }
+
+        /// <summary>
+        /// Find a class by Id
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <returns>The class with the given id or null if not found</returns>
+        public Class GetById(int classId)
+        {
+            return _db.Classes.Find(classId);
+        }
+
+        /// <summary>
+        /// Returns all classes taught by the given teacher
+        /// </summary>
+        /// <param name="teacherId">The id of the teacher</param>
+        /// <returns>List of classes taught by the teacher</returns>
+        public IEnumerable<Class> GetClassesForTeacher(int teacherId)
+        {
+            return _db.Classes
+                .Where(@class => @class.Teacher.Id == teacherId)
+                .Include(c => c.Lectures)
+                .ToList();
```

Comment:
It returns an IQueryable (which is an IEnumerable). I was thinking about eager vs. lazy loading, but I guess it doesn't matter much for our purposes